pilot/pkg/xds/xds_cache.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	"sync"
+
+	"github.com/golang/protobuf/ptypes/any"
+
+	"istio.io/istio/pkg/config/host"
+)
+
+// Cache interface defines a store for caching XDS responses
+// Note this is currently only for EDS, and will need some modifications to support other types
+// All operations are thread safe
+type Cache interface {
+	Insert(key EndpointBuilder, value *any.Any)
```

Comment:
May be better to start introducing a `CacheKey` interface with method to get key and EndpointBuilder implements it so that it is easier going forward?

Reply:
Yeah probably so, I'll change this

Reply:
+1 - EndpointBuilder as a key sounds strange. Maybe just rename EndpointBuilder ? Is it really a builder ?